src/librustc_mir/borrow_check/nll/region_infer/mod.rs

```diff
@@ -1112,7 +1097,7 @@ impl<'tcx> RegionInferenceContext<'tcx> {
         let relevant_constraint = self.constraints
             .iter_enumerated()
             .filter_map(|(i, constraint)| {
-                if self.liveness_constraints.contains(constraint.sub, elem) {
+                if !self.liveness_constraints.contains(constraint.sub, elem) {
```

Comment:
(it would be nice if this one-line bugfix were attached to the commit that generalized `blame_constraint`)